### ISSUE TITLE
Change lightgrep scanners to use lookaround assertions

### ISSUE THREAD

**Issue by jonstewart** [simsong/bulk_extractor#163]
Lightgrep has some limited support for lookaround assertions now, which would be very useful in almost all of the lightgrep-based scanners.

**Comment by simsong** [comment]
What is the advantage  of doing this? And how do we test it? And don’t these things dramatically  increase processing time?

**Comment by jonstewart** [comment]
1. Eliminates a bunch of fiddly post-search code.
2. With unit tests first, then on data.
3. It depends on the expression, input data, and regex engine. We can measure but I doubt that it'll have a significant impact. Lightgrep's uses a two-byte filter at the tightest rank within the automaton that's before the minimum length of a hit, and a one character forward assertion shouldn't impact that much/at all.

**Comment by simsong** [comment]
Makes sense. I would really like to have some performance comparisons between the flex scanners and lightgrep. We should be sure to put that in the paper.
